runtime-sdk/modules/rofl-market/src/lib.rs

```diff
@@ -626,8 +626,10 @@ impl<Cfg: Config> Module<Cfg> {
 
         let mut provider = state::get_provider(body.provider).ok_or(Error::ProviderNotFound)?;
         Self::ensure_caller_is_scheduler_app(&provider)?;
-        let mut instance =
-            state::get_instance(body.provider, body.id).ok_or(Error::InstanceNotFound)?;
+        let mut instance = match state::get_instance(body.provider, body.id) {
+            Some(instance) => instance,
+            None => return Ok(()), // Already removed.
```

Comment:
Why is this better? Again, user has no idea which road was taken. Or is expected for the caller to observe events also?

Reply:
The user is the scheduler and I don't see a case where this would matter, especially since multiple scheduler instances could issue removal for the same instance. The scheduler internally uses randomized intervals to reduce probability of conflicts, but I don't see a reason why a failure here is the right thing in case the instance already does not exist. An instance with the same identifier will never exist again.

The scheduler could also just ignore the error in such cases and it would be fine, so we can also leave it as-is.

Reply:
Ok, you're right, it makes sense to just fail here.

Reply:
> The scheduler could also just ignore the error in such cases 

I think you should return an error and the scheduler should ignore it.